lib/backup/azremote/azblob.go

```diff
@@ -278,14 +273,58 @@ func (fs *FS) DeleteFile(filePath string) error {
 	}
 
 	path := fs.Dir + filePath
-	bc := fs.clientForPath(path)
-	if err != nil {
-		return err
+	return fs.delete(path)
+}
+
+func (fs *FS) delete(path string) error {
+	if *common.DeleteAllObjectVersions {
+		return fs.deleteObjectWithGenerations(path)
+	}
+	return fs.deleteObject(path)
+}
+
+func (fs *FS) deleteObjectWithGenerations(path string) error {
+	pager := fs.client.NewListBlobsFlatPager(&azblob.ListBlobsFlatOptions{
+		Prefix: &path,
+		Include: azblob.ListBlobsInclude{
+			Versions: true,
+		},
+	})
+
+	ctx := context.Background()
+	for pager.More() {
+		resp, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("cannot list blobs at %s (remote path %q): %w", path, fs.Container, err)
+		}
+
+		for _, v := range resp.Segment.BlobItems {
+			var c *blob.Client
+			// Either versioning is disabled or we are deleting the current version
+			if v.VersionID == nil || (v.VersionID != nil && v.IsCurrentVersion != nil && *v.IsCurrentVersion) {
+				c = fs.client.NewBlobClient(*v.Name)
```

Comment:
Azure API looks as always weird.

 Should we group new clients by version name or its fine to create new client each time?

 Also, should we close client at the end of delete operation?

Reply:
> Should we group new clients by version name or its fine to create new client each time?
> Also, should we close client at the end of delete operation?

This `client` is just an API wrapper which stores several params and nothing else(no open connections or something similar), so I think it is ok to create a new one.